### PR TITLE
Add concrete steps in doc/cjdns.md to easily find a friend

### PR DESCRIPTION
**Pull request by jonatack**
and improve the informational links.  CJDNS functions with a friend-of-a-friend topology and a key hurdle to getting started is to find a public peer and set up an outbound connection to it. This update makes doing it much easier for people getting started. 

Credit to Vasil Dimov for an [IRC suggestion in October 2021](https://www.erisian.com.au/bitcoin-core-dev/log-2021-10-04.html#l-469) and to stickies-v for IRC discussions this week and the [testing guide](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/23.0-Release-Candidate-Testing-Guide) that led me to redo these steps, provide feedback at https://github.com/bitcoin/bitcoin/issues/24706 and refine the added documentation here.

### CODE REVIEW

File: doc/cjdns.md [hunk 1]
```diff
@@ -23,17 +24,35 @@ somewhat centralized. I2P connections have a source address and I2P is slow.
...
+One quick way to accomplish these two steps is to query for available public
+peers on [Hyperboria](https://github.com/hyperboria) by running the following
+
+```
+git clone https://github.com/hyperboria/peers hyperboria-peers
+cd hyperboria-peers
+./testAvailable.py
+```
+
+and to choose a friend from among the peers returned in the
```
Comment (stickies-v) [inline doc/cjdns.md:48]: I think it would be prudent to recommend adding a few, say, 3-5 friends instead of just one?

File: doc/cjdns.md [hunk 2]
```diff
@@ -23,17 +24,35 @@ somewhat centralized. I2P connections have a source address and I2P is slow.
...
+`hyperboria-peers/<continent/<country>/<city>` directories.
+
+You can then copy/paste its credentials into the relevant IPv4 or IPv6
+"connectTo" JSON object in the `cjdroute.conf` file in your CJDNS directory and
```
Comment (stickies-v) [inline doc/cjdns.md:52]: Maybe this is platform specific, but I'm not sure there always is a "CJDNS directory"? Perhaps you could phrase it as "in the `cjdroute.conf` file you created in the setup phase"?

File: doc/cjdns.md [hunk 3]
```diff
@@ -23,17 +24,35 @@ somewhat centralized. I2P connections have a source address and I2P is slow.
...
+peers on [Hyperboria](https://github.com/hyperboria) by running the following
+
+```
+git clone https://github.com/hyperboria/peers hyperboria-peers
+cd hyperboria-peers
+./testAvailable.py
+```
+
+and to choose a friend from among the peers returned in the
+`hyperboria-peers/<continent/<country>/<city>` directories.
```
Comment (stickies-v) [inline doc/cjdns.md:49]: Phrasing is slightly ambiguous I think. Maybe you could rephrase that the script prints the location of the credentials and the pinging result, and then user can go fetch the credentials from that location and copy/paste into config? Just highlights a bit more the user action required to actually go into the folders based on script output.

**Review by stickies-v — COMMENTED** [review]
Looks good, I like the additional info. Even though there's some information duplication, including the friend finding process should make things a lot easier for new users.

Left a few comments, but beyond that pretty much good to go for me.

File: doc/cjdns.md (hunk 2 above)
Comment (jonatack) [reply doc/cjdns.md:52]: Done

File: doc/cjdns.md (hunk 3 above)
Comment (jonatack) [reply doc/cjdns.md:49]: Done

**Comment by jonatack** [comment]
Thank you for the feedback, @stickies-v. Updated.

File: doc/cjdns.md [hunk 4]
```diff
@@ -23,17 +24,37 @@ somewhat centralized. I2P connections have a source address and I2P is slow.
...
+cd hyperboria-peers
+./testAvailable.py
+```
+
+For each peer, the `./testAvailable.py` script prints the filename of the peer's
+credentials followed by the ping result.
+
+Choose one or several peers, copy their credentials from their respective files,
+paste them into the relevant IPv4 or IPv6 "connectTo" JSON object in the
+`cjdroute.conf` you created in step ["1. Generate a new configuration
```
Comment (stickies-v) [inline doc/cjdns.md:53]: nit: missing word "file" after `cjdroute.conf`

Comment (jonatack) [reply doc/cjdns.md:53]: thanks, added

**Review by stickies-v — APPROVED** [review]
ACK 3c2f7d8

**Review by brunoerg — APPROVED** [review]
ACK 3c2f7d883531a7d21e53446b25e53909226f67a4

I followed it when testing v23 and the steps added here worked fine and was helpful.

File: doc/cjdns.md [hunk 5]
```diff
@@ -23,17 +24,37 @@ somewhat centralized. I2P connections have a source address and I2P is slow.
 CJDNS is fast but does not hide the sender and the recipient from intermediate
 routers.
 
-## Installing CJDNS and connecting to the network
+## Installing CJDNS and finding a friend to connect to the network
```
Comment (laanwj) [inline doc/cjdns.md:27]: Should it still be called a friend if it's simply a random active public peer selected from a list? :smile:

Comment (jonatack) [reply doc/cjdns.md:27]: Kind of like a dating app? :smile:

**Comment by laanwj** [comment]
ACK 3c2f7d883531a7d21e53446b25e53909226f67a4

**Comment by jonatack** [comment]
Thanks for the reviews and feedback; updated with the following change.
```diff
diff --git a/doc/cjdns.md b/doc/cjdns.md
index e0e31ef130..b69564729f 100644
--- a/doc/cjdns.md
+++ b/doc/cjdns.md
@@ -24,7 +24,7 @@ somewhat centralized. I2P connections have a source address and I2P is slow.
 CJDNS is fast but does not hide the sender and the recipient from intermediate
 routers.
 
-## Installing CJDNS and finding a friend to connect to the network
+## Installing CJDNS and finding a peer to connect to the network
 
 To install and set up CJDNS, follow the instructions at
 https://github.com/cjdelisle/cjdns#how-to-install-cjdns.
```

**Review by stickies-v — APPROVED** [review]
re-ACK [6a02355](https://github.com/bitcoin/bitcoin/commit/6a02355ae9e830ea40b5408acc564e9eb1da1378) even though I wasn't opposed to the "friend" terminology since it's the language CJDNS seems to use to denominate the peers you connect to directly in general. Not worth bikeshedding over though.

**Comment by dunxen** [comment]
ACK 6a02355

**Review by dunxen — APPROVED** [review]

**Review by lsilva01 — APPROVED** [review]
Strong ACK 6a02355

I skipped this step when setting up the CJDNS for the first time.
So I think it's good to reinforce this in the documentation.

**Pull request merged**